cmds/webboot/webboot.go

```diff
@@ -51,69 +47,60 @@ func (d *DownloadOption) exec(uiEvents <-chan ui.Event) (menu.Entry, error) {
 	}
 
 	fpath := filepath.Join("/tmp", filename)
-	if err = download(link, fpath); err != nil {
-		return nil, err
-	}
-
-	var iso *ISO = &ISO{
-		label: filename,
-		path:  fpath,
+	// if download link is not valid, ask again until the link is rights
+	for err = download(link, fpath); err != nil; err = download(link, fpath) {
```

Comment:
Why do we need to do `err = download(link, fpath)` twice? Hm would it work to do the loop this way

```
derr := download(link, fpath)
for derr != nil {
  // display derr
 // get input
  derr = download(link, fpath)
}
```

Reply:
Ok that would works too thanks. the reason I write download(link, fpath) twice here because the for loop's first statement is a initialization, second is the judgement; third is the statement will run at the begin of every loop.